src/server/scripts/Outland/TempestKeep/arcatraz/boss_harbinger_skyriss.cpp

```diff
@@ -283,9 +283,15 @@ class boss_harbinger_skyriss_illusion : public CreatureScript
         {
             boss_harbinger_skyriss_illusionAI(Creature* creature) : ScriptedAI(creature) { }
 
-            void Reset() override { }
+            void Reset() override 
+            {
+                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+            }
 
-            void EnterCombat(Unit* /*who*/) override { }
+            void EnterCombat(Unit* /*who*/) override 
+            {
+                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
```

Comment:
Reset() is called on spawn, so by the time they enter combat it'll be already gone, so repeating it here might have no point.